src/main/java/frc/robot/subsystems/ArmPID.java

```diff
@@ -0,0 +1,122 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import frc.robot.RobotMap;
+
+import edu.wpi.first.wpilibj.AnalogPotentiometer;
+import edu.wpi.first.wpilibj.command.PIDSubsystem;
+import edu.wpi.first.wpilibj.Spark;
+
+public class ArmPID extends PIDSubsystem {
+
+  public double scaleFactor = 50; 
+
+  public Spark m_arm = new Spark(RobotMap.m_arm);
+
+  public AnalogPotentiometer armPoten = new AnalogPotentiometer(RobotMap.potentiometer, scaleFactor);
+
+  public double zeroPoint = 0;
+
+  private int index    = 0;
+  public double output = 0; 
```

Comment:
It seems output and target aren't used in your code. These variables are likely used for setting the motor "output" and the "target" encoder position for your PID control loop to adjust the motor speeds to maintain those preset positions.

I'd suggest renaming these variables to be more clear once we figure out what they are supposed to do.